decidim-stats/README.md

```diff
@@ -0,0 +1,82 @@
+# Decidim::Stats
+
+## Production setup
+
+This module adds demographic statistics about the participants.
+
+In order to run it, please check you have these environment variables set:
+
+```text
+STATS_MINIMUM_COUNT=5
+STATS_EMAILS=foo@example.org,bar@example.org
+```
+
+`STATS_MINIMUM_COUNT` sets the lower limit of data to be exported. Any stats with a total below that number will be ignored. The default value is `5`.
+
+`STATS_EMAILS` is a list of the emails the stats file will be sent to. If you eed to send it to more than one email, please separate them by commas: `foo@bar.com,yaz@zez.com`
+
+Assuming you have access to the Heroku application, use this command to set the environment variables:
+
+```text
+heroku config:set STATS_MINIMUM_COUNT=5 STATS_EMAILS=foo@example.org,bar@example.org
+```
+
+The command to generate the stats file is this:
+
+```text
+rake stats:generate
+```
+
+This command will send an email to thre specified email addresses with the stats. You can use [Heroku Scheduler](https://devcenter.heroku.com/articles/scheduler#scheduling-jobs) to generate the file automatically every night.
+
+## Stats CSV format
+
+The CSV file has these columns:
+
+```text
+participatory_space_type,participatory_space_id,component_id,action,metric_type,metric_name,total
+participatory_processes,2,4,comment,age_group,20-24,6
+```
+
+## Development guide
+
+The module has different pieces:
+
+- `Decidim::Stats::Runner`, which calculates the stats. Internally, it uses:
+  - `Decidim::Stats::Performers`, which define how to group users by different metrics. Each performer groups users by different criteria. Only users verified via the Census are considered. Current `performers` are:
+    - `AgeGroup`, grouping by birth date
+    - `District`, grouping by district
+    - `Gender`, grouping by user sex/gender
+  - `Decidim::Stats:Actions`, which define the different actions users perform and we want to track. All actions receive a `Decidim::Component` instance and a group of performers.
+    - `Comment` counts how many performers have commented any resource of the given component
+    - `Endorsement` counts how many performers have endorsed a resource of the given component
+    - `Follow` counts how many performers have followed a resource of the given component
+    - `ProposalCreation` counts how many performers have created a proposal in the given component
+    - `ProposalVote` counts how many performers have voted a proposal of the given component
+- `Decidim::Stats::CsvExporter`, which generates a temporary CSV file from the stats data
+- `Decidim::Stats::StatsMailer`, which emails the CSV file to the emails specified in the environment variable.
+- All the steps are run by the `Decidim::Stats::StatsJob`
+
+Apart from these actions, the module calculates an `interactions` statistics that is the number of performers that have performed any of the previously listed actions in a given component.
```

Comment:
Question: I understand that the count that each action does is the `total` column in the CSV. But there is no `interactions` column. So `interactions` is another kind of action? if yes, shouldn't we add it to the previous list? if no, can you explain?

Reply:
Interactions is an action that counts the number of users that have performed ANY of the rest of the actions. Since it depends on the result of the other actions, this is calculated from the `Runner` class, instead of being a normal action. That's why I didn't put it in the list, because it doesn't have a class but is calculated once all the other ones are finished.